extensions/reactive-routes/deployment/src/test/java/io/quarkus/vertx/web/context/DuplicatedContextTest.java

```diff
@@ -0,0 +1,129 @@
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.http.HttpClientResponse;
+import io.vertx.core.http.HttpMethod;
+import io.vertx.core.impl.EventLoopContext;
+import io.vertx.core.impl.WorkerContext;
+import io.vertx.ext.web.RoutingContext;
+
+/**
+ * Verify that reactive routes are called on duplicated contexts and that they handled them properly.
+ */
+public class DuplicatedContextTest {
+
+    @RegisterExtension
+    static final QuarkusUnitTest config = new QuarkusUnitTest()
+            .withApplicationRoot((jar) -> jar
+                    .addClasses(MyRoutes.class));
+
+    @Test
+    public void testThatRoutesAreCalledOnDuplicatedContext() {
+        // Creates a bunch of requests that will be executed concurrently.
+        // So, we are sure that event loops are reused.
+        List<Uni<Void>> unis = new ArrayList<>();
+        for (int i = 0; i < 500; i++) {
+            String uuid = UUID.randomUUID().toString();
+            unis.add(Uni.createFrom().item(() -> {
+                String resp = get("/context/" + uuid).asString();
+                Assertions.assertEquals(resp, "OK-" + uuid);
+                return null;
+            }));
+        }
+
+        Uni.join().all(unis).andFailFast()
+                .runSubscriptionOn(Infrastructure.getDefaultExecutor())
+                .await().atMost(Duration.ofSeconds(10));
+    }
+
+    @Test
+    public void testThatBlockingRoutesAreCalledOnDuplicatedContext() {
+        // Creates a bunch of requests that will be executed concurrently.
+        // So, we are sure that event loops are reused.
+        List<Uni<Void>> unis = new ArrayList<>();
+        for (int i = 0; i < 500; i++) {
+            String uuid = UUID.randomUUID().toString();
+            unis.add(Uni.createFrom().item(() -> {
+                String resp = get("/context-blocking/" + uuid).asString();
+                Assertions.assertEquals(resp, "OK-" + uuid);
+                return null;
+            }));
+        }
+
+        Uni.join().all(unis).andFailFast()
+                .runSubscriptionOn(Infrastructure.getDefaultExecutor())
+                .await().atMost(Duration.ofSeconds(10));
+
+    }
+
+    @ApplicationScoped
+    public static class MyRoutes {
+
+        @Inject
+        Vertx vertx;
+
+        @Route(path = "/context/:id", methods = Route.HttpMethod.GET)
+        void get(RoutingContext ctx) {
+            Assertions.assertTrue(Thread.currentThread().getName().contains("vert.x-eventloop"));
+            process(ctx);
+        }
+
+        @Route(path = "/context-blocking/:id", methods = Route.HttpMethod.GET)
+        @Blocking
+        void getBlocking(RoutingContext ctx) {
+            Assertions.assertFalse(Thread.currentThread().getName().contains("vert.x-eventloop"));
+            process(ctx);
+        }
+
+        private void process(RoutingContext ctx) {
+            Context context = Vertx.currentContext();
+            Assertions.assertFalse(context instanceof EventLoopContext);
```

Comment:
Would `context.isOnEventLoopThread()` still return `true`?

Reply:
in the case of blocking... yes.
I know it's confusing. It's because of `executeBlocking` which keeps your event loop context. 

I've opened a PR on Vert.x to get a better way to check if we are on a duplicated context.